packages/react-router/src/server/lowQualityTransactionsFilterIntegration.ts

```diff
@@ -0,0 +1,35 @@
+import { type Client, type Event, type EventHint, defineIntegration, logger } from '@sentry/core';
+import type { NodeOptions } from '@sentry/node';
+
+/**
+ * Integration that filters out noisy http transactions such as requests to node_modules, favicon.ico, @id/
+ *
+ */
+
+function _lowQualityTransactionsFilterIntegration(options: NodeOptions): {
+  name: string;
+  processEvent: (event: Event, hint: EventHint, client: Client) => Event | null;
+} {
+  const matchedRegexes = [/GET \/node_modules\//, /GET \/favicon\.ico/, /GET \/@id\//];
+
+  return {
+    name: 'LowQualityTransactionsFilter',
+
+    processEvent(event: Event, _hint: EventHint, _client: Client): Event | null {
+      if (event.type !== 'transaction' || !event.transaction) {
+        return event;
+      }
+
+      if (matchedRegexes.some(regex => event.transaction?.match(regex))) {
```

Comment:
```suggestion
      if (matchedRegexes.some(regex => event.transaction.match(regex))) {
```

l: no need for that I suppose, as we check agains this being undefined above? Not sure, maybe TS also wants this to be stored in a variable before we check it 😬